File: AssetSync/src/AssetSync.Core/Models/SyncRunSummary.cs
namespace AssetSync.Core.Models;

public class SyncRunSummary
{
    public string SyncRunId { get; set; } = Guid.NewGuid().ToString();
    public DateTimeOffset StartedAtUtc { get; set; }
    public DateTimeOffset CompletedAtUtc { get; set; }
    public bool DryRun { get; set; }
    public int Created { get; set; }
    public int Updated { get; set; }
    public int Skipped { get; set; }
    public int Errors { get; set; }
    public bool SnipeItReachable { get; set; }
    public bool IntuneReachable { get; set; }
    public bool IruReachable { get; set; }
}

File: AssetSync-macOS/AssetSyncCore/Sources/AssetSyncCore/Models/SyncRunSummary.swift
import Foundation

public struct SyncRunSummary: Sendable {
    public var syncRunId: String = UUID().uuidString
    public var startedAtUtc: Date = Date()
    public var completedAtUtc: Date = Date()
    public var dryRun: Bool = false
    public var created: Int = 0
    public var updated: Int = 0
    public var skipped: Int = 0
    public var errors: Int = 0
    public var snipeItReachable: Bool = false
    public var intuneReachable: Bool = false
    public var iruReachable: Bool = false

    public init() {}
}
